### ISSUE TITLE
False positive for B031 with generator used in method

### ISSUE THREAD

**Issue by 153957** [charliermarsh/ruff#3885]
Used command:
`ruff --select B031 test.py`

Used ruff:
`ruff 0.0.261`

Code snippet:
```python
z = []
for _, x in groupby([]):
    z.append([1 for _ in x])  # B031
```

Error:
```Using the generator returned from `itertools.groupby()` more than once will do nothing on the second usage```

Can change the code like this to prevent the error:
```python
z = []
for _, x in groupby([]):
    y = [1 for _ in x]
    z.append(y)
```

**Comment by dhruvmanila** [comment]
Oh, right. Actually, there's another bug due to which the updated code doesn't raise a warning. I'll fix both of them today.
